tests/e2e/test_e2e_skip_trained.py

```diff
@@ -0,0 +1,244 @@
+        X_dict={"data": X_dev},
+        Y_dict={"label1": Y_dev},
+    )
+
+    test_dataset = EmmentalDataset(
+        name="synthetic",
+        X_dict={"data": X_test},
+        Y_dict={"label1": Y_test},
+    )
+
+    task_to_label_dict = {"task1": "label1"}
+
+    train_dataloader = EmmentalDataLoader(
+        task_to_label_dict=task_to_label_dict,
+        dataset=train_dataset,
+        split="train",
+        batch_size=10,
+    )
+    dev_dataloader = EmmentalDataLoader(
+        task_to_label_dict=task_to_label_dict,
+        dataset=dev_dataset,
+        split="valid",
+        batch_size=10,
+    )
+    test_dataloader = EmmentalDataLoader(
+        task_to_label_dict=task_to_label_dict,
+        dataset=test_dataset,
+        split="test",
+        batch_size=10,
+    )
+
+    # Create task
+    def ce_loss(task_name, immediate_output_dict, Y, active):
+        module_name = f"{task_name}_pred_head"
+        return F.cross_entropy(
+            immediate_output_dict[module_name][0][active], (Y.view(-1))[active]
+        )
+
+    def output(task_name, immediate_output_dict):
+        module_name = f"{task_name}_pred_head"
+        return F.softmax(immediate_output_dict[module_name][0], dim=1)
+
+    task_metrics = {"task1": ["accuracy"], "task2": ["accuracy"]}
+
+    class IdentityModule(nn.Module):
+        def __init__(self):
+            """Initialize IdentityModule."""
+            super().__init__()
+
+        def forward(self, input):
+            return {"out": input}
+
+    tasks = [
+        EmmentalTask(
+            name=task_name,
+            module_pool=nn.ModuleDict(
+                {
+                    "input_module0": IdentityModule(),
+                    "input_module1": nn.Linear(2, 8),
+                    f"{task_name}_pred_head": nn.Linear(8, 2),
+                }
+            ),
+            task_flow=[
+                {
+                    "name": "input",
+                    "module": "input_module0",
+                    "inputs": [("_input_", "data")],
+                },
+                {
+                    "name": "input1",
+                    "module": "input_module1",
+                    "inputs": [("input", "out")],
+                },
+                {
+                    "name": f"{task_name}_pred_head",
+                    "module": f"{task_name}_pred_head",
+                    "inputs": [("input1", 0)],
+                },
+            ],
+            module_device={"input_module0": -1},
+            loss_func=partial(ce_loss, task_name),
+            output_func=partial(output, task_name),
+            action_outputs=[
+                (f"{task_name}_pred_head", 0),
+                ("_input_", "data"),
+                (f"{task_name}_pred_head", 0),
+            ]
+            if task_name == "task2"
+            else None,
+            scorer=Scorer(metrics=task_metrics[task_name]),
+            require_prob_for_eval=True if task_name in ["task2"] else False,
```

Comment:
I think this was a snippet taken from somewhere else. As you just loop through "task1" you remove this line and the "if task_name == 'task2'" piece. It just makes things a little more confusing.

Reply:
Good catch! Thanks!